import axios from "axios"

const api = "http://localhost:8581";

export function getCustomer(jwt){
    const header={
        Authorization: `Bearer ${jwt}`
    }
    axios.get(`${api}/api/customer`,{
        Authorization: `Bearer ${jwt}`
    })
    .then((res)=>{
        console.log(res)
    })
    .catch((error)=>{
        console.error('Ошибка при отправке запроса:', error.response ? error.response.data : error.message);
        if (error.response) {
            console.log(`Ошибка: ${error.response.data.message}`);
        } else {
            console.log('Произошла ошибка при подключении к серверу.');
        }
    })
}

export function getCustomerServiceTypeOrg(){
    const headers={

    }
    axios.get(`${api}/api/customer/service_type/organizations`,headers)
    .then(()=>{

    })
    .catch(()=>{
        console.error('Ошибка при отправке запроса:', error.response ? error.response.data : error.message);
        if (error.response) {
            console.log(`Ошибка: ${error.response.data.message}`);
        } else {
            console.log('Произошла ошибка при подключении к серверу.');
        }
    })
}

export function getCustomerServiceRequest(){
    const headers={

    }
    axios.get(`${api}/api/customer/service_requests`,headers)
    .then(()=>{

    })
    .catch(()=>{
        console.error('Ошибка при отправке запроса:', error.response ? error.response.data : error.message);
        if (error.response) {
            console.log(`Ошибка: ${error.response.data.message}`);
        } else {
            console.log('Произошла ошибка при подключении к серверу.');
        }
    })
}

export function updateCustomer(){
    const body={

    }
    const headers={

    }
    axios.put(`${api}/api/customer/update`,body,headers)
    .then(()=>{

    })
    .catch(()=>{
        console.error('Ошибка при отправке запроса:', error.response ? error.response.data : error.message);
        if (error.response) {
            console.log(`Ошибка: ${error.response.data.message}`);
        } else {
            console.log('Произошла ошибка при подключении к серверу.');
        }
    })
}
export function deleteCustomer(){
    const headers={

    }
    const body={

    }
    axios.delete(`${api}/api/organization/`,body,headers)
    .then(()=>{

    })
    .catch(()=>{
        console.error('Ошибка при отправке запроса:', error.response ? error.response.data : error.message);
        if (error.response) {
            console.log(`Ошибка: ${error.response.data.message}`);
        } else {
            console.log('Произошла ошибка при подключении к серверу.');
        }
    })
}